bindings/python/setup.py

```diff
@@ -18,11 +18,6 @@
 # Whether to create libarrow symlinks on posix systems.
 CREATE_LIBARROW_SYMLINKS = os.environ.get("MONGO_CREATE_LIBARROW_SYMLINKS", "1")
 
-# Whether to add the GLIBCXX override.
-# Arrow's manylinux{2010, 2014} binaries are built with gcc <= 4.8 which predates CXX11 ABI
-# - https://uwekorn.com/2019/09/15/how-we-build-apache-arrows-manylinux-wheels.html
-ADD_GLIBCXX_OVERRIDE = os.environ.get("MONGO_ADD_GLIBCXX_OVERRIDE", "1")
```

Comment:
This was what was causing the issue.  The pyarrow wheels are now using manylinux_2_28, which has gcc 12.